src/PrestaShopBundle/Form/Admin/Catalog/Category/AbstractCategoryType.php

```diff
@@ -88,27 +101,35 @@ public function __construct(
         array $customerGroupChoices,
         FeatureInterface $multistoreFeature,
         ConfigurationInterface $configuration,
-        Router $router
+        Router $router,
+        TacticianCommandBusAdapter $queryBus
     ) {
         parent::__construct($translator, $locales);
 
         $this->customerGroupChoices = $customerGroupChoices;
         $this->multistoreFeature = $multistoreFeature;
         $this->configuration = $configuration;
         $this->router = $router;
+        $this->queryBus = $queryBus;
     }
 
     /**
      * {@inheritdoc}
      */
     public function buildForm(FormBuilderInterface $builder, array $options)
     {
+        $coverImage = $thumbnailImage = $menuThumbnailImages = $categoryId = null;
         $disableMenuThumbnailsUpload = false;
         if (isset($options['id_category'])) {
-            $disableMenuThumbnailsUpload = $options['disable_menu_thumbnails_upload'];
+            $categoryId = (int) $options['id_category'];
+            $editableCategory = $this->queryBus->handle(new GetCategoryForEditing((int) $options['id_category']));
```

Comment:
The query shouldn't be executed inside the abstract type but in the `CategoryFormDataProvider` (actually I believe it already is). So the data you're missing here should either be passed as data directly via the provider Or via options if you want to keep this separated from the data

If you need some "advanced options" based on the result of the query you can rely on the `FormOptionsProviderInterface` which has a similar behaviour as the `FormDataProviderInterface` except it builds an array of options that is passed to the form type when it's built

You can see an example here https://github.com/PrestaShop/PrestaShop/blob/3cd5ea8d86ab4578f914648cf423ad885d47c6e3/src/Core/Form/IdentifiableObject/OptionProvider/ProductFormOptionsProvider.php#L38 and you can then configure it in the form builder service definition:
https://github.com/PrestaShop/PrestaShop/blob/28cb9109c2027b4f8061548623c76cccb60e91eb/src/PrestaShopBundle/Resources/config/services/core/form/form_builder.yml#L202

The good thing is that the `getOptions` method also contains the `$data` generated via the data provider so you can use some of the data to generate your options